signer/src/transaction_coordinator.rs

```diff
@@ -392,15 +440,33 @@ where
             }
 
             match operation_result {
-                Some(wsts::state_machine::OperationResult::SignTaproot(signature)) => {
-                    return Ok(signature)
-                }
+                Some(res) => return Ok(res),
                 None => continue,
-                Some(_) => return Err(Error::UnexpectedOperationResult),
             }
         }
     }
 
+    /// Check whether or not we need to run DKG
+    /// 
+    /// This function checks for the existence of a
+    /// [`RotateKeysTransaction`] in the database, and one does not exist
```

Comment:
```suggestion
    /// Check whether or not we need to run DKG
    /// 
    /// This function first checks for the existence of a
    /// [`RotateKeysTransaction`] row in the database. If one exists, then DKG
    /// has already been completed.
    ///
    /// If not, it then proceeds to check for DKG shares in the database for
    /// the current signer aggregate key. If DKG shares exist but there is no
    /// [`RotateKeysTransaction`] then one needs to be created, otherwise if
    /// no shares exist then DKG needs to be run.
```

Reply:
perfect, yeah I forgot to update this.